main.cpp

```diff
@@ -73,418 +73,233 @@ compile/link/run to make sure you don't have any errors
 If you need to view an example, see: https://bitbucket.org/MatkatMusic/pfmcpptasks/src/master/Projects/Project4/Part7Example.cpp
 */
 
-
 #include <cmath>
 #include <functional>
-
-struct FloatType;
-struct IntType;
-struct DoubleType;
+#include <memory>
+//#include <type_traits> 
+//#include <limits> 
 
 struct Point
 {
     Point(float a, float b) : x(a), y(b) {}
-    Point(FloatType& a, FloatType& b);
-    Point(DoubleType& a, DoubleType& b);
-    Point(IntType& a, IntType& b);
+    
+    //constructor to take numeric types
+    template<typename T>
+    Point(T& a, T& b) : Point(static_cast<float>(a), static_cast<float>(b) ) { }
+
+    template<typename T> 
+    Point& multiply( T& m)
+    {
+        return multiply(static_cast<float>(m));
+    }
 
     Point& multiply(float m)
     {
         x *= m;
         y *= m;
         return *this;
     }
-    
-    Point& multiply( FloatType& m );
-    Point& multiply( DoubleType& m );
-    Point& multiply( IntType& m );
 
-    void toString();
+    void toString()
+    {
+        std::cout << "( " << x << ", " << y << " )" << std::endl;
+    }
 
 private:
     float x{0}, y{0};
 };
 
-
-// float UDT 
-struct FloatType
+//template class
+template<typename T>
+struct Numeric
 {
+    using ValueType = T;
     public:
```

Comment:
`struct` is public by default
- [ ] remove `public`

your `using` alias should be `Type` in preparation for Part 8
 - [ ] revise